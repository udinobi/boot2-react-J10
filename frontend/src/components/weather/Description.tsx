import React from "react"

import styled from "styled-components"

import { MediumText } from "../lib/StyledText"

import { Weather } from "../../store/weather/types"

export interface Data {
    weather: Weather[]
}

const camelCase = (s: string) =>
    s.replace(
        /\w\S*/g,
        (txt: string) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
    )

const StyledDescription = styled(MediumText)`
    color: orange;
    margin: 6px 0 12px;
`

export default (props: Data) => 
    <StyledDescription>
        { props.weather.length > 0
            ? camelCase(props.weather[0].description)
            : ""
        }
    </StyledDescription>
