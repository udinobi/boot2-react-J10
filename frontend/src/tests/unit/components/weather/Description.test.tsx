import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import Description, { Data } from "../../../../components/weather/Description"

describe("--> Snapshot for Components > weather > 'Description'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            weather: []
        }
    })

    it("matches", () => {
        const wrapper = shallow(<Description {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
