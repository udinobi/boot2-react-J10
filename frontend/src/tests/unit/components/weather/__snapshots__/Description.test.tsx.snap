// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'Description' matches 1`] = `<Styled(Styled(styled.div)) />`;
